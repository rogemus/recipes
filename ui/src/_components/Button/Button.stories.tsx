import { fn } from "@storybook/test";
import { Button } from "./Button";

import type { Meta, StoryObj } from "@storybook/react";

const meta: Meta<typeof Button> = {
  component: Button,
  title: "Button",
  args: {
    label: "Button",
    onClick: fn(),
  },
};

export default meta;

type Story = StoryObj<typeof Button>;

export const Default: Story = {};

export const Disabled: Story = {
  args: {
    disabled: true,
  },
};
