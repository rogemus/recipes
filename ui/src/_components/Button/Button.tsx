import type { FC } from "react";
import type { ButtonProps } from "./Button.types";
import styles from "./Button.module.css";

const Button: FC<ButtonProps> = ({
  label,
  testId = "Button",
  disabled = false,
  type = "button",
  onClick,
}) => {
  return (
    <button
      type={type}
      className={styles.button}
      data-testid={testId}
      {...(disabled && { disabled })}
      {...(onClick && { onClick })}
    >
      {label}
    </button>
  );
};

export { Button };
