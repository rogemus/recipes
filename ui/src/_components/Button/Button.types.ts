import { Comp } from "@/_models";
import { MouseEventHandler } from "react";

export type ButtonProps = {
  disabled?: boolean;
  label: string;
  type?: HTMLButtonElement["type"];
  onClick?: MouseEventHandler<HTMLButtonElement>;
} & Comp;
