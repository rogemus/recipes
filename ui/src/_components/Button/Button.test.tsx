import { expect, test, describe, vi } from "vitest";
import { fireEvent, render, screen } from "@testing-library/react";

import { Button } from "./Button";

import type { ButtonProps } from "./Button.types";

const defaultProps: ButtonProps = {
  label: "Test label",
};

describe("@component/Button", () => {
  test("renders component correctly", () => {
    const { container } = render(<Button {...defaultProps} />);

    expect(container).toMatchSnapshot();
  });

  test("renders disabled component correctly", () => {
    const { container } = render(<Button {...defaultProps} disabled />);

    expect(container).toMatchSnapshot();
  });

  test("onClick triggered", () => {
    const mockClick = vi.fn();
    render(<Button {...defaultProps} onClick={mockClick} />);

    const button = screen.getByTestId("Button");
    fireEvent.click(button);

    expect(mockClick).toHaveBeenCalledOnce();
  });
});
