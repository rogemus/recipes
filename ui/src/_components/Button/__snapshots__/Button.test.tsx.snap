// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`@component/Button > renders component correctly 1`] = `
<div>
  <button
    class="_button_f0af68"
    data-testid="Button"
    type="button"
  >
    Test label
  </button>
</div>
`;

exports[`@component/Button > renders disabled component correctly 1`] = `
<div>
  <button
    class="_button_f0af68"
    data-testid="Button"
    disabled=""
    type="button"
  >
    Test label
  </button>
</div>
`;
